src/runtime/startup/c/wrapper.c

```diff
@@ -596,31 +622,39 @@ __attribute__((constructor)) static void init(void) {
 
     close(output[1]);
 
-    fds.fd = output[0];
-    fds.events = POLLIN;
-    fds.revents = 0;
+    fds[0].fd = output[0];
+    fds[0].events = POLLIN;
+    fds[0].revents = 0;
+
+    fds[1].fd = master_socket[0];
+    fds[1].events = POLLIN;
+    fds[1].revents = 0;
 
     singularity_message(DEBUG, "Wait C and JSON runtime configuration from scontainer stage 1\n");
 
-    while ( poll(&fds, 1, -1) >= 0 ) {
-        if ( fds.revents & POLLIN ) {
+    while ( poll(fds, 2, -1) >= 0 ) {
+        if ( fds[0].revents & POLLIN ) {
             int ret;
             singularity_message(DEBUG, "Receiving configuration from scontainer stage 1\n");
-            if ( (ret = read(output[0], &config, sizeof(config))) != sizeof(config) ) {
+            if ( (ret = read(fds[0].fd, &config, sizeof(config))) != sizeof(config) ) {
                 singularity_message(ERROR, "Failed to read C configuration stdout pipe: %s\n", strerror(errno));
                 exit(1);
             }
             if ( config.jsonConfSize >= MAX_JSON_SIZE) {
                 singularity_message(ERROR, "JSON configuration too big\n");
                 exit(1);
             }
-            if ( (ret = read(output[0], json_stdin, config.jsonConfSize)) != config.jsonConfSize ) {
+            if ( (ret = read(fds[0].fd, json_stdin, config.jsonConfSize)) != config.jsonConfSize ) {
                 singularity_message(ERROR, "Failed to read JSON configuration from stdout pipe: %s\n", strerror(errno));
                 exit(1);
             }
             json_stdin[config.jsonConfSize] = '\0';
+            break;
+        }
+        /* TODO: pass file descriptors from stage 1 over unix socket */
```

Comment:
I assume this is just put here temporarily until we need to use this socket?

Reply:
Yes

Reply:
👍